sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -165,7 +165,10 @@ statement
     | CREATE (OR REPLACE)? TEMPORARY? FUNCTION (IF NOT EXISTS)?
         identifierReference AS className=stringLit
         (USING resource (COMMA resource)*)?                            #createFunction
-    | DROP TEMPORARY? FUNCTION (IF EXISTS)? identifierReference      #dropFunction
+    | DROP TEMPORARY? FUNCTION (IF EXISTS)? identifierReference        #dropFunction
+    | DECLARE (OR REPLACE)? VARIABLE?
+        multipartIdentifier dataType? variableDefaultExpression?      #createVariable
```

Comment:
```suggestion
        multipartIdentifier dataType? variableDefaultExpression?       #createVariable
```

Reply:
Looking at the `dataType`, one of the possibilities is an `identifier`:

```
dataType
    : complex=ARRAY LT dataType GT                              #complexDataType
    ...
    | type (LEFT_PAREN INTEGER_VALUE
      (COMMA INTEGER_VALUE)* RIGHT_PAREN)?                      #primitiveDataType
    ;

type
    : BOOLEAN
    ...
    | unsupportedType=identifier
    ;
```

So if `DEFAULT` or `EQ` match against this, could there be parser ambiguity between the `dataType` and the `variableDefaultExpression`, since both are optional? If so, should we add an explicit check for the ambiguous case(s) to make sure they don't fall through the cracks?